src/dolphin/unwrap/_unwrap_3d.py

```diff
@@ -108,3 +120,38 @@ def _map_date_str_to_file(
     # None is special case for reference epoch
     date_str_to_file[first_date] = None
     return date_str_to_file
+
+
+def _create_conncomps_from_mask(
+    temporal_coherence_file: PathOrStr,
+    temporal_coherence_threshold: float,
+    unw_filenames: Sequence[PathOrStr],
+    dilate_by: int = 21,
+) -> list[Path]:
+    arr = io.load_gdal(temporal_coherence_file, masked=True)
+    good_pixels = arr > temporal_coherence_threshold
+    strel = np.ones((dilate_by, dilate_by))
+    # All "1" pixels will be spread out and have (approximately) 1.0 surrounding pixels
+    # Threshold back to be a binary image
+    good_pixels_dilated = signal.fftconvolve(good_pixels, strel, mode="same") > 0.95
```

Comment:
that was what i started checking, but when you dilate by more than a few pixels, it's very slow
```
In [23]: %time ndimage.label( ndimage.binary_dilation( temp_coh > 0.6 , np.ones((11, 11))), np.ones((3, 3)))[1]
CPU times: user 12.2 s, sys: 964 ms, total: 13.2 s
Wall time: 21 s
Out[23]: 9679

In [24]: %time ndimage.label( ndimage.binary_dilation( temp_coh > 0.6 , np.ones((15, 15))), np.ones((3, 3)))[1]
CPU times: user 19.9 s, sys: 1.4 s, total: 21.3 s
Wall time: 35.7 s
Out[24]: 3223

In [25]: %time ndimage.label( ndimage.binary_dilation( temp_coh > 0.6 , np.ones((21, 21))), np.ones((3, 3)))[1]
CPU times: user 31.6 s, sys: 1.32 s, total: 32.9 s
Wall time: 1min 2s
Out[25]: 433
```

Reply:
whereas 

```
In [38]: %time fftconvolve(goods, np.ones((21, 21)) )

CPU times: user 1.38 s, sys: 446 ms, total: 1.82 s
Wall time: 1.85 s
```

Reply:
I see, thanks.

As a general practice, I think that any time you find yourself thinking something like "well I would implement this using {{simple, intuitive method}} but it turns out that's not the best approach here because {{reasons}} so instead I'll use {{different, non-obvious approach}}", you should probably consider explaining the reasoning behind the non-obvious implementation in a comment.

Maybe we could add a comment here explaining that we don't want to use binary dilation because that scales poorly with the size of the dilation kernel, whereas FFT convolution is fast even for larger dilations.